### PR TITLE
Add report-only post-round evaluation to the R2LPL runner

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## What this PR does

This PR adds **automatic post-round evaluation** to the R2LPL lifelong training runner (`rlvr.autoresearch.tools.run_lifelong_r2lpl_rounds`), plus several correctness fixes to the training loop itself.

### Problem

The R2LPL runner trains a model over several rounds (mine failures → generate repaired trajectories → retrain). Until now there was no built-in way to tell whether a round actually made the model better or quietly made it worse somewhere else. Regressions (e.g. the model no longer stopping for a pedestrian, or new collisions on roads it used to handle fine) only showed up much later, in manual evaluation.

### Post-round evaluation ("guards")

A new optional `guards` section in the workflow config. When set, the runner evaluates the starting model once (reference row) and then every round's checkpoint, writing the metrics to `r2lpl_round_NNN/guards/guard_metrics.json` and into `round_summary.json`:

- **Held-out closed-loop test**: the checkpoint drives a FROZEN set of held-out 8-second road segments in closed loop, and rule violations are counted per type — collisions, road-border crossings, and disagreement with the recorded human driver. The disagreement count is split by direction: the model kept going where the human stopped (fail to stop) vs. the model stayed put where the human drove off (fail to take off).
- **Stopping/resuming benchmark**: on a frozen set of scenes where the recorded driver stops for an interaction, the checkpoint's deterministic plan is checked for: does it stop at all, how late, does it resume when the driver resumed, and how much distance it drives vs. the recording.
- **Optional full closed-loop probe** via `valid_predictor_closed_loop` (expensive, off by default).

The evaluation is **report-only**: it never accepts or rejects checkpoints. The README documents a simple recipe for picking the best round from the resulting tables. (An automatic accept/reject gate with rollback was tried during development and deliberately removed: retrying a rejected round just re-rolls the same data and never recovered, and a hard threshold gate discards good checkpoints whenever the threshold is imperfect.)

The README also documents how to build the frozen test set correctly: a seeded random sample of segments from the whole corpus, never a contiguous slice (a contiguous slice measures only one stretch of one route).

### Training-loop fixes

- **Mixed scene formats crashed training**: raw logged scenes (used as "anchor" scenes to preserve normal driving competence) store neighbor futures with 3 columns, repaired scenes with 4. A batch containing both crashed the data loader. The runner now converts anchor scenes to the 4-column format before training.
- **`replay_memory.capacity` must now be set explicitly** — the previous silent default (200) is far too small for large datasets and would quietly evict earlier rounds' training scenes.
- Config validation moved to startup: an anchor-slot repair variant without its prototype file, a too-small candidate count, or an unknown variant name now fail immediately instead of after a full mining phase.

### New repair-candidate generator

`anchor_fan_16` generation variant: repair candidates are exactly the model's deterministic plan plus 16 prototype-path-guided variations (via the anchor-following guidance), K=17 — no noise sampling. The scripted stop/slow-down candidate is still injected for driver-disagreement events.

## Test plan

- 145 unit tests (runner wiring end-to-end with mocked phases, config validation, metric derivation, the 3-col/4-col conversion, the new generation variant).
- Several full end-to-end training campaigns (multiple rounds × multiple epochs) on a ~13k-scene corpus, exercising mining, repair with the new generator, anchored training, and the per-round evaluation; the evaluation tables correctly tracked improvements and caught regressions (including a failure mode shifting from "doesn't stop" to "doesn't take off" over rounds).

### CODE REVIEW

File: rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py [hunk 1]
```diff
@@ -1017,6 +1052,39 @@ def _sample(pool: list[str], n: int) -> list[str]:
...
+    out: list[str] = []
+    for path in paths:
+        src = Path(path)
+        with np.load(src, allow_pickle=True) as loaded:
+            future = loaded["neighbor_agents_future"]
+            if future.shape[-1] == 4:
+                out.append(str(src))
+                continue
+            data = dict(loaded)
+        data["neighbor_agents_future"] = _future_to_4col(future)
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py]: Fixed in d42d2c7b: dropped allow_pickle (these NPZs are pure numeric arrays) and added an explicit channel-count check — anything other than 3 or 4 channels now raises with the offending path instead of failing inside the converter.

File: rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py [hunk 2]
```diff
@@ -1067,6 +1135,331 @@ def _validate_anchor_config(cfg: dict[str, Any]) -> None:
...
+    if closed_loop is not None:
+        if guards.get("closed_loop_npz_root") is None:
+            raise ValueError("guards.closed_loop knobs are set without closed_loop_npz_root")
+        unknown_cl = sorted(set(closed_loop) - _GUARD_CLOSED_LOOP_KEYS)
+        if unknown_cl:
+            raise ValueError(
+                f"guards.closed_loop has unknown keys {unknown_cl}; "
+                f"allowed: {sorted(_GUARD_CLOSED_LOOP_KEYS)}"
+            )
+
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py:1257]: Fixed in d42d2c7b: the closed_loop sub-section now tolerates "_"-prefixed comment keys like every other guards section.

File: rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py [hunk 3]
```diff
@@ -1067,6 +1135,331 @@ def _validate_anchor_config(cfg: dict[str, Any]) -> None:
...
+        "-m",
+        "valid_predictor_closed_loop",
+        "--model_path",
+        str(Path(model_path).resolve()),
+        "--npz_root",
+        str(Path(guards["closed_loop_npz_root"]).resolve()),
+    ]
+    for key, value in sorted(dict(guards.get("closed_loop") or {}).items()):
+        cmd.extend([f"--{key}", str(value)])
+    return cmd
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py]: Fixed in d42d2c7b: the probe command builder strips "_"-prefixed keys, so direct configs can carry template-style comments without leaking --_comment* flags (the workflow path already stripped them at parse time). Pinned by a test.

File: rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py [hunk 4]
```diff
@@ -1067,6 +1140,407 @@ def _validate_anchor_config(cfg: dict[str, Any]) -> None:
...
+    slots needs a prototype library, and K must cover det + every fixed slot —
+    otherwise the failure surfaces only at repair time, after a full mining
+    phase.
+    """
+    from rlvr.generation_variants import get_variant
+
+    repair_cfg = dict(cfg.get("repair_config") or {})
+    variant_name = str(repair_cfg.get("variant", ""))
+    if not variant_name:
+        return
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py:1178]: Fixed: a null/empty variant is treated as not set instead of coercing to the literal string "None" (pinned by tests).

File: rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py [hunk 5]
```diff
@@ -1067,6 +1140,407 @@ def _validate_anchor_config(cfg: dict[str, Any]) -> None:
...
+    """
+    guards = cfg.get("guards")
+    policy = str(cfg.get("checkpoint_policy", "latest"))
+    if policy == "composite":
+        raise ValueError(
+            "checkpoint_selection_rule='composite' was removed: automatic "
+            "reject/rollback never recovered a round (retries just re-roll the same "
+            "distribution). Guards are report-only — use 'latest' and select "
+            "checkpoints from the per-round guard tables"
+        )
```
[the remark this answers is not in the document]
Comment (danielsanchezaran) [reply rlvr/autoresearch/tools/run_lifelong_r2lpl_rounds.py:1216]: Fixed: the error now names both spellings — rounds.checkpoint_selection_rule (workflow configs) and checkpoint_policy (direct configs).
